client/map_verifier.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"crypto"
+	"fmt"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keys/der"
+	"github.com/google/trillian/merkle"
+	"github.com/google/trillian/merkle/hashers"
+
+	tcrypto "github.com/google/trillian/crypto"
+)
+
+// MapVerifier verifies protos produced by the Trillian Map.
+type MapVerifier struct {
+	Hasher hashers.MapHasher
+	PubKey crypto.PublicKey
+}
+
+// NewMapVerifierFromTree creates a new MapVerifier.
+func NewMapVerifierFromTree(config *trillian.Tree) (*MapVerifier, error) {
+	if got, want := config.TreeType, trillian.TreeType_MAP; got != want {
```

Comment:
I thought this got, want style was to be restricted to test code, where it is idiomatic.
In production code it just seems to obscure the intent a little.

Reply:
In production code the got, want style can be helpful if the got and want values are reused inside the if block

Reply:
ok